evm/p2p/lightchain.py

```diff
@@ -331,28 +336,41 @@ def _test():
         NETWORK_ID = MAINNET_NETWORK_ID
     DemoLightChain = LightChain.configure(
         'DemoLightChain',
-        privkey=ecies.generate_privkey(),
         vm_configuration=MAINNET_VM_CONFIGURATION,
         network_id=NETWORK_ID,
     )
 
     chaindb = BaseChainDB(LevelDB(args.db))
+    if args.local_geth:
+        peer_pool = LocalGethPeerPool(LESPeer, chaindb, NETWORK_ID, ecies.generate_privkey())
+    else:
+        peer_pool = PeerPool(LESPeer, chaindb, NETWORK_ID, ecies.generate_privkey())
     try:
         chaindb.get_canonical_head()
     except CanonicalHeadNotFound:
         # We're starting with a fresh DB.
-        chain = DemoLightChain.from_genesis_header(chaindb, GENESIS_HEADER)
+        chain = DemoLightChain.from_genesis_header(chaindb, GENESIS_HEADER, peer_pool)
     else:
         # We're reusing an existing db.
-        chain = DemoLightChain(chaindb)
+        chain = DemoLightChain(chaindb, peer_pool)
```

Comment:
Note to self.  This `try/except/else` is a bit cumbersome and I found myself replicating it in like 2-3 places so far.  going to try and figure out a more elegant/expressive/intuitive way to implement this logic.

Reply:
Maybe something like?

```py
chain = DemoLightChain.load(chaindb, peer_pool, default_genesis=ROPSTEN_GENESIS_HEADER)
```